src/main/java/org/opensearch/knn/index/query/KNNQuery.java

```diff
@@ -37,9 +38,13 @@ public class KNNQuery extends Query {
 
     private final String field;
     private final float[] queryVector;
+    @Getter
+    private final byte[] byteQueryVector;
```

Comment:
can we use <T> here rather than having 2 separate variables for query vectors and byteQueryVector.

Reply:
Could you elaborate? Do you want it to be `KNNQuery<T>` or having another `KNNQueryByte` , or `private final Object queryVector`?

Reply:
I am more of thinking this as `KNNQuery<T>` and not another `KNNQueryByte`